src/vs/editor/contrib/parameterHints/parameterHintsWidget.ts

```diff
@@ -312,8 +312,14 @@ export class ParameterHintsWidget extends Disposable implements IContentWidget {
 		} else if (Array.isArray(param.label)) {
 			return param.label;
 		} else {
-			const idx = signature.label.lastIndexOf(param.label);
-			return idx >= 0
+			const idx = Math.max(
```

Comment:
This fix seems specific to JavaScript style signatures. How about looking for the last match of a regular expression such as `\b${param.label}\b` which would check for the param label surrounded by word boundaries? Not perfect but that way `param` would not match `parameter`

Reply:
Thanks for your feedback. Let me give that a shot

Reply:
Updated. Let me know what you think